feat(landing-page): new 2021 landing page

mattnolting opened the pull request:
New landing page

Docs: 
- https://docs.google.com/document/d/16mrk2gORQUgVp13EbJN5_GbkKDDkRdaPprQaogZY1ww/edit
- https://xd.adobe.com/view/e9237ffa-089b-471a-b697-1882d52e9b78-bc70/screen/204eea27-e82b-4c3c-8c3b-4a6b71b0ead9/specs/

<img width="891" alt="Screen Shot 2021-04-16 at 7 59 13 AM" src="https://user-images.githubusercontent.com/5385435/115021249-b9368c00-9e89-11eb-8b95-e38aea3b787e.png">

<img width="888" alt="Screen Shot 2021-04-16 at 7 59 21 AM" src="https://user-images.githubusercontent.com/5385435/115021260-bc317c80-9e89-11eb-9b07-ef3f8d7c8b0f.png">

<img width="899" alt="Screen Shot 2021-04-16 at 7 59 32 AM" src="https://user-images.githubusercontent.com/5385435/115021266-be93d680-9e89-11eb-8dcf-35a59b9024bd.png">

epwinchell commented on src/routes/Landing.scss:
```diff
@@ -117,9 +119,10 @@ sup.ins-c-rball { @include rem('font-size', 10px); }
 }
 
-// Adjust PF layout to be able to display witouth header
+// Adjust PF layout to be able to display whithout header
```
without :)

epwinchell reviewed (approved):
@mattnolting Very nice! Tested in FF, Safari and Chrome at all breakpoints.

Hyperkid123 commented:
@mattnolting can you fix up the tests, please? I looked through the code briefly and I was wondering how do we decide what links open a new window/tab? I see some links to redhat.com are external and some are not.

Also, do we have some mocks or a preview or did you make it from scratch?

karelhala commented on src/components/HeroIcon.js:
```diff
@@ -0,0 +1,16 @@
+import './HeroIcon.scss';
+
+const HeroIcon = ({ src, alt }) => (
+  <div className="ins-c-hero__title-icon">
```
This is using flex layout right? Might be more reasonable to use the layout
```JSX
import { Flex, FlexItem } from '@patternfly/react-core';
const HeroIcon = (props) => (
  <Flex className="ins-c-hero__title-icon">
    <img {...props} aria-hidden="true" />
  </Flex>
);
```

The only thing that will be still required is the `width`.

karelhala commented on src/components/marketing/icon__build-deploy.svg:
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
```
No need to include SVGs in this repository we have https://github.com/RedHatInsights/frontend-assets repository to host these files.

karelhala commented on src/layout/FooterMarketing.js:
```diff
@@ -0,0 +1,67 @@
+import './FooterMarketing.scss';
+
+const FooterTraditional = () => {
+  return (
+    <React.Fragment>
```
Extra `return` and `React.Fragment` No need to use these as you have just one `PageSection`

karelhala commented on src/layout/FooterMarketing.scss:
```diff
@@ -0,0 +1,42 @@
+
+  *,
+  a {
+    color: #aaa;
```
I don't really like custom colors, can you use `--pf-global--palette--black-500` or `--pf-global--palette--black-600` for this one? Or are they just way too off for this place?

karelhala commented on src/layout/FooterMarketing.scss:
```diff
@@ -0,0 +1,42 @@
+  --pf-c-page__main-section--PaddingBottom: var(--pf-global--spacer--md);
+  --pf-c-page__main-section--PaddingLeft: var(--pf-global--spacer--md);
+
+  background-color: #1a1a1a;
```
I think the closes PF color to this one is `--pf-global--palette--black-900` can we use it here?

karelhala commented on src/layout/FooterMarketing.scss:
```diff
@@ -0,0 +1,42 @@
+  }
+
+  a:hover {
+    color: #fff;
```
Same as above, can we use PF var color `--pf-global--palette--black-100`

karelhala commented on src/layout/Hero.js:
```diff
@@ -1,38 +1,71 @@
+        >
+          {heroContent.primaryCTAtext}
+        </Button>
+        {/* Activate when support is available
+        <Button
+          className="ins-m-marketing"
+          variant="secondary"
+          component="a"
+          isLarge
+          href={heroContent.secondaryCTAurl}
+        >
+          {heroContent.secondaryCTAtext}
+        </Button> */}
```
No need to keep commented out code. Github tracks the history for us.

karelhala commented on src/layout/Hero.scss:
```diff
@@ -2,20 +2,71 @@
+        width: var(--ins-c-hero--before--Width);
+        content: '';
+        opacity: var(--ins-c-hero--before--Opacity);
+        background-image: #{url('https://raw.githubusercontent.com/RedHatInsights/frontend-assets/0331b39e1a370e5f1c243452330df61f4e1b57bc/src/background-images/bg__hero.svg')};
```
You don't have to use `githubusercontent` as URL, just `https://cloud.redhat.com/apps/frontend-assets/src/background-images/bg__hero.svg` is the URL correct @ryelo ?

ryelo replied:
Correct

karelhala commented on src/layout/Marketing.scss:
```diff
@@ -1,45 +1,172 @@
+        border-radius: 3px;
+
+        // Primary
+        --pf-c-button--m-primary--Color: #fff;
+        --pf-c-button--m-primary--BackgroundColor: #ee0000;
+
+        // Secondaery
+        --pf-c-button--m-secondary--Color: #151515;
+        --pf-c-button--m-secondary--after--BorderColor: #151515;
+        --pf-c-button--m-secondary--hover--after--BorderColor: #151515;
+        --pf-c-button--m-secondary--BackgroundColor: transparent;
```
Can we use PF colors for these values?

karelhala reviewed (commented):
Looking good! I have just a few nitpicks about using custom colors instead of PF ones and perhaps moving SVGs to frontend-assets repo and using the full URL for them.

ryelo commented:
@karelhala not trying to speak for Matt here, but I believe this page is actually using branded Red Hat colors + assets (not Patternfly), so there may be some inconsistencies with the color mapping.

This could also totally be wrong 😄

ryelo commented on src/consts.js:
```diff
@@ -1,41 +1,287 @@
-import openShiftMarketing from './components/marketing/openShiftMarketing.svg';
-import ansibleMarketing from './components/marketing/ansibleMarketing.svg';
+export const productGridHeader = 'Build your cloud on open source';
+export const productGridLoginText = 'Login in to the console';
```
`Login in`

mattnolting commented:
> @karelhala not trying to speak for Matt here, but I believe this page is actually using branded Red Hat colors + assets (not Patternfly), so there may be some inconsistencies with the color mapping.
> 
> This could also totally be wrong 😄

That's _pretty much_ correct. I've updated all colors we currently provide w/exception for marketing specific values and added notes for those custom assignments.

Hyperkid123 commented:
@mattnolting I've added a commit that fixes up the tests and build. There is a list of changes

- some of the unAuthed assertions were irrelevant due to the technology data change. I have removed obsolete assertions but kept as much as possible
- One major issue was the different Modal root element. I've mocked the root element to the Landing test file
- Fixed a bunch of linting errors
- Added safe accessor to nested objects which might be undefined

Please check if the visuals did not change in any way (they should not)

mattnolting commented:
@Hyperkid123 Looks awesome, thank you!!

john-dupuy commented on src/layout/GetStarted.js:
```diff
@@ -0,0 +1,48 @@
+        <Button
+          className="ins-m-marketing"
+          variant="primary"
+          isLarge
```
Is this the main login button? If so, can we add an ouiaId prop to this one? QE can use it to easily locate the button within the DOM. 

It is enough to just add a prop like
```
ouiaId="primary-login-button"
```

For the top right login button, it'd be nice if we had some ouiaId like
```
ouiaId="top-right-login-button"
```

ryelo replied:
@john-dupuy top right one is coming from chrome

john-dupuy replied:
oh yeah! let me go add an id to that one

john-dupuy replied:
Thanks!

ryelo commented:
Yeah this LGTM when you can add that OUIA attribute

mattnolting commented:
@ryelo @john-dupuy Updated
